src/modelinstance.cpp

```diff
@@ -237,6 +238,48 @@ Status ModelInstance::loadOVCNNNetwork() {
     return StatusCode::OK;
 }
 
+Status ModelInstance::loadOVCNNNetworkUsingCustomLoader() {
+    spdlog::debug("Try reading model using a custom loader");
+    try {
+        char* xmlBuffer = NULL;
+        int xmlLen = 0;
+        char* binBuffer = NULL;
+        int binLen = 0;
+
+        SPDLOG_INFO("loading CNNNetwork for model:{} basepath:{} <> {} version:{}", getName(), getPath(), this->config.getBasePath().c_str(), getVersion());
+        auto& customloaders = ovms::CustomLoaders::instance();
+        auto customLoaderInterfacePtr = customloaders.find(customLoaderName);
+
+        if (customLoaderInterfacePtr == nullptr) {
+            SPDLOG_INFO("Loader {} is not in loaded customloaders list", customLoaderName);
+            throw std::invalid_argument("customloader not exisiting");
+        }
+
+        CustomLoaderStatus res = customLoaderInterfacePtr->loadModel(this->config.getName(),
+            this->config.getBasePath(),
+            getVersion(),
+            this->config.getCustomLoaderOptionsConfigStr(),
+            &xmlBuffer, &xmlLen, &binBuffer, &binLen);
+        if ((res == CustomLoaderStatus::MODEL_LOAD_ERROR) || (res == CustomLoaderStatus::INTERNAL_ERROR)) {
+            return StatusCode::INTERNAL_ERROR;
+        }
+
+        // Ravikb--> Todo check return make empty blob
+        std::string strModel(xmlBuffer);
+        std::vector<uint8_t> weights(&binBuffer[0], &binBuffer[binLen]);
+        network = std::make_unique<InferenceEngine::CNNNetwork>(engine->ReadNetwork(strModel,
+            make_shared_blob<uint8_t>({Precision::U8, {weights.size()}, C}, weights.data())));
+        // deleting the buffers got from custom loader
+        delete[] xmlBuffer;
```

Comment:
See my earlier comment with using unique_ptr. In case of exception you have a leak.

Reply:
Changed the code to use vector by reference and got rid of pointers